In [17]:
import numpy as np

def rule1(dA,c):
    dQ_1=np.abs(c)*dA
    return dQ_1

dA=3
c=2
dQ_1=rule1(dA,c)

print("dQ_1=",dQ_1)

def rule2(c,m,A,dA):
    dQ_2 = np.abs(c*m*A**(m-1))*dA
    return dQ_2

c=1
m=2
A=4
dA=4
dQ_2=rule2(c,m,A,dA)

print("dQ_2=",dQ_2)



def rule3(dA, dB):
    dQ_3 = np.sqrt(dA**2+dB**2)
    return dQ_3

A=1.0
B=0.3
dA=0.0005
dB=0.0005
Q_3=A+B
dQ_3=rule3(dA,dB)

print("Q_3=", Q_3)
print("dQ_3=", dQ_3)

def rule4(A,B,C,D,m,n,o,p,dA,dB,dC,dD,Q):
    dQ_4 = np.abs(Q)*np.sqrt((m*dA/A)**2+(n*dB/B)**2+(o*dC/C)**2+(p*dD/D)**2)
    return dQ_4

A=1
dA=1
m=2
B=2
dB=2
n=2
C=1
dC=0
o=0
D=1
dD=0
p=0
Q=1
dQ_4=rule4(A,B,C,D,m,n,o,p,dA,dB,dC,dD,Q)
#leave capital variables as 1 to avoid a divide by zero error#



print("dQ_4=", dQ_4)

c =  205.2
dQ_1= 6
dQ_2= 32
Q_3= 1.3
dQ_3= 0.0007071067811865475
dQ_4= 2.8284271247461903


In [5]:
import numpy as np

x=np.array([1,2,3,4,5])
xmean=np.average(x)
xstdev=np.std(x)
print(xmean)
print(xstdev)

3.0
1.4142135623730951


In [4]:
import numpy as np

def rule4(A,B,C,D,m,n,o,p,dA,dB,dC,dD,Q):
    dQ_4 = np.abs(Q)*np.sqrt((m*dA/A)**2+(n*dB/B)**2+(o*dC/C)**2+(p*dD/D)**2)
    return dQ_4

A=98.35
dA=.5
m=.5
B=1
dB=0
n=0
C=1
dC=0
o=0
D=1
dD=0
p=0
Q=3.43
dQ_4=rule4(A,B,C,D,m,n,o,p,dA,dB,dC,dD,Q)
#leave capital variables as 1 to avoid a divide by zero error#

print("dVi=", dQ_4)

dVi= 0.008718861209964414


My uncertainty from the ballistic pendulum lab was 0.01 due to significant figures. I calculated the exact same value using excel for my report however as the value calculated via the python functions.

In [5]:
import numpy as np

x=np.array([1.1, 1.3, 1.4, 0.9, 0.95, 1.05])
xmean=np.average(x)
xstdev=np.std(x)
print(xmean)
print(xstdev)

1.1166666666666667
0.17950549357115014


The numpy functions returned a mean value of 1.1 and a standard deviation of 0.2 when considering significant figures. When not considering significant figures the mean was about 1.1167 and the standard deviation was about 0.1795.

If $Q = cA^mB^n$ where $c$, $m$, and $n$ are constants, then $\frac{\delta Q}{|{Q}|} = \sqrt{(m\frac{\delta A}{A})^2 + (n\frac{\delta B}{B})^2} $. I accidentally did rule four instead of rule three.  

The equation for the uncertainty of $V_i$ from the ballistic pendulum lab is as follows: $ \delta V_i = V_i\sqrt{.5(\frac{\delta y}{y})^2} $.